pkg/reconciler/kubernetes/tektonhub/tektonhub.go

```diff
@@ -0,0 +1,580 @@
+	}
+
+	th.Status.MarkPreReconcilerComplete()
+
+	exist, err = checkIfInstallerSetExist(ctx, r.operatorClientSet, version, th, apiInstallerSet)
+	if err != nil {
+		return err
+	}
+
+	if !exist {
+		th.Status.MarkApiInstallerSetNotAvailable("API installer set not available")
+		apiLocation := filepath.Join(hubDir, "api")
+		err := r.applyManifest(ctx, apiLocation, th, apiInstallerSet, version, "hub-api")
+		if err != nil {
+			return err
+		}
+	}
+
+	err = r.checkComponentStatus(ctx, th, apiInstallerSet)
+	if err != nil {
+		th.Status.MarkApiInstallerSetNotAvailable(err.Error())
+		r.enqueueAfter(th, 10*time.Second)
+		return err
+	}
+
+	th.Status.MarkApiInstallerSetAvailable()
+
+	// Create the API route based on platform
+	if err := r.extension.PostReconcile(ctx, th); err != nil {
+		return err
+	}
+
+	th.Status.MarkPostReconcilerComplete()
+
+	return nil
+}
+
+// Validate DB are present on the cluster. If DB secrets are present and all the keys don't
+// have values then update the remaining values with default values. If the DB secret
+// is not present then create a new secret with default values.
+func (r *Reconciler) validateOrCreateDBSecrets(ctx context.Context, th *v1alpha1.TektonHub) error {
+	logger := logging.FromContext(ctx)
+	dbKeys := []string{"POSTGRES_DB", "POSTGRES_USER", "POSTGRES_PASSWORD", "POSTGRES_PORT"}
+
+	th.Status.MarkDbDependencyInstalling("db secrets are being added into the namespace")
+
+	dbSecret, err := r.getSecretForHub(ctx, th.Spec.Db.DbSecretName, namespace, dbKeys)
+	if err != nil {
+		newDbSecret := createDbSecret(th.Spec.Db.DbSecretName, namespace, dbSecret)
+		if apierrors.IsNotFound(err) {
+			_, err = r.kubeClientSet.CoreV1().Secrets(namespace).Create(ctx, newDbSecret, metav1.CreateOptions{})
+			if err != nil {
+				logger.Error(err)
+				th.Status.MarkDbDependencyMissing(fmt.Sprintf("%s secret is missing", th.Spec.Db.DbSecretName))
+				return err
+			}
+			return nil
+		}
+		if err == errKeyMissing {
+			_, err = r.kubeClientSet.CoreV1().Secrets(namespace).Update(ctx, newDbSecret, metav1.UpdateOptions{})
+			if err != nil {
+				logger.Error(err)
+				th.Status.MarkDbDependencyMissing(fmt.Sprintf("%s secret is missing", th.Spec.Db.DbSecretName))
+				return err
+			}
+		} else {
+			logger.Error(err)
+			return err
+		}
+	}
+
+	return nil
+}
+
+// TektonHub expects API secrets to be created before installing Tekton Hub API
+func (r *Reconciler) validateApiDependencies(ctx context.Context, th *v1alpha1.TektonHub) error {
+	logger := logging.FromContext(ctx)
+	apiSecretKeys := []string{"GH_CLIENT_ID", "GH_CLIENT_SECRET", "JWT_SIGNING_KEY", "ACCESS_JWT_EXPIRES_IN", "REFRESH_JWT_EXPIRES_IN", "GHE_URL"}
+	apiConfigMapKeys := []string{"CONFIG_FILE_URL"}
+
+	th.Status.MarkApiDependencyInstalling("checking for api secrets in the namespace and creating the ConfigMap")
+
+	_, err := r.getSecretForHub(ctx, th.Spec.Api.ApiSecretName, namespace, apiSecretKeys)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
```

Comment:
I think we are not creating secrets anywhere 
we should create the secrets here